### PR TITLE
[docs] Add docs about running tests in CI

### PR DESCRIPTION
**Pull request by dpakach**
Related https://github.com/owncloud/ocis/issues/303#issuecomment-644510374

Add docs about the CI tests structure.

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -150,3 +150,20 @@ If you want to work on a specific issue
...
+
+## Acceptance Tests In CI
+In the CI we run the existing API tests from oc10 in various repos. Since the tests are in the oc10 core repo, we use the commit IDs from core to indicate which version of tests to run in the CI. The commit IDs need to be frequently updated to have all the new tests running because new tests are constantly being added in the core repo.
+
+We run the API tests from core in following repos.
+
+### 1. ocis Repo
+In the `owncloud/ocis` repo the tests from the core are executed by cloning a specific version of the test runner. The commit id of the version can be specified in the `.drone.star` file in ocis.
```
Comment (kiranparajuli589) [inline docs/testing.md:160]:
```suggestion
In the `owncloud/ocis` repo the tests from the core are executed by cloning a specific version of the test runner. The commit ID of the version can be specified in the `.drone.star` file in OCIS.
```

File: docs/testing.md [hunk 2]
```diff
@@ -150,3 +150,20 @@ If you want to work on a specific issue
...
+```
+apiTests(ctx, 'master', 'a3cac3dad60348fc962d1d8743b202bc5f79596b')
+```
+if the version you want to run is on the different branch from master, you also need to change the branch name.
```
Comment (kiranparajuli589) [inline docs/testing.md:164]:
```suggestion
If the version you want to run is on the different branch from master, you also need to change the branch name.
```

File: docs/testing.md [hunk 3]
```diff
@@ -150,3 +150,20 @@ If you want to work on a specific issue
...
+
+In order to check if new tests are compatible with OCIS, after changing the commit id and the branch name, we can create a draft PR in ocis which trigger's the CI and we can see the result there.
```
Comment (kiranparajuli589) [inline docs/testing.md:166]:
```suggestion
In order to check if new tests are compatible with OCIS, after changing the commit ID and the branch name, we can create a draft PR in OCIS which triggers the CI and we can see the result there.
```

File: docs/testing.md [hunk 4]
```diff
@@ -150,3 +150,20 @@ If you want to work on a specific issue
...
+
+### 2. ocis-reva Repo
+We follow the same approach in the ocis-reva repo too. In order to run the API tests in CI we use commit Ids which can be changed in the `.drone.star` file just like we did in ocis repo.
```
Comment (kiranparajuli589) [inline docs/testing.md:169]:
```suggestion
We follow the same approach in the ocis-reva repo too. In order to run the API tests in CI, we use commit IDs which can be changed in the `.drone.star` file just like we did in OCIS repo.
```

File: docs/testing.md [hunk 5]
```diff
@@ -132,13 +132,11 @@ If you want to work on a specific issue
...
-    For that change this line in the `acceptance-tests` section
+    For that change this line in the `main` function
 
-    `'git clone -b master --depth=1 https://github.com/owncloud/core.git /srv/app/testrunner',`
+        apiTests(ctx, 'master', 'a3cac3dad60348fc962d1d8743b202bc5f79596b')
 
-    to clone your core branch e.g.
-
-    `'git clone -b fixRevaIssue122 --depth=1 https://github.com/owncloud/core.git /srv/app/testrunner',`
+    In place of master use your branch name. Also make sure to change the commit ID to the hash of the latest commit in your branch or any other commits you want to test.
```
Comment (phil-davis) [inline docs/testing.md:139]: Or you can delete the commit id:
`apiTests(ctx, 'my-branch', '')`

and the CI will use the current head of the branch in core.

**Pull request merged**